File: php/src/Model/InviteWorkerIdentity.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * Input for emailing a worker a set-your-PIN link. Creates the worker too when it does not exist yet, which the result reports as `workerCreated`.
 *
 * Fluent builder: set what you need, then hand it to the client. {@see self::toArray()} omits
 * every field that was never set, because the API treats an absent field as "leave unchanged"
 * and an explicit null as "clear it".
 */
final class InviteWorkerIdentity
{
    public function __construct(
        private readonly string $email,
        private ?string $workerId = null,
        private ?string $label = null,
        /** @var list<string>|null */
        private ?array $tags = null,
        /** @var list<WorkerSkillAssignment>|null */
        private ?array $skills = null,
        /** @var list<string>|null */
        private ?array $teamIds = null,
    ) {
    }

    public function workerId(string $workerId): self
    {
        $this->workerId = $workerId;
        return $this;
    }

    public function label(string $label): self
    {
        $this->label = $label;
        return $this;
    }

    public function tags(array $tags): self
    {
        $this->tags = $tags;
        return $this;
    }

    public function skills(array $skills): self
    {
        $this->skills = $skills;
        return $this;
    }

    public function teamIds(array $teamIds): self
    {
        $this->teamIds = $teamIds;
        return $this;
    }

    /**
     * Serialise for the wire, omitting unset fields.
     *
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        return [
            'email' => $this->email,
        ] + Json::compact([
            'workerId' => $this->workerId,
            'label' => $this->label,
            'tags' => $this->tags,
            'skills' => Json::each($this->skills),
            'teamIds' => $this->teamIds,
        ]);
    }
}

File: java/src/main/java/io/fivexer/sdk/model/InviteWorkerIdentity.java
package io.fivexer.sdk.model;

import io.fivexer.sdk.internal.Json;
import java.util.List;

/**
 * Input for emailing a worker a set-your-PIN link. Creates the worker too when it does not exist yet, which the result reports as {@code workerCreated}.
 *
 * <p>Fluent builder: set what you need, then hand it to the client. {@link #toJson()} omits
 * every field that was never set, because the API treats an absent field as "leave unchanged"
 * and an explicit null as "clear it".
 */
public class InviteWorkerIdentity {
    private String email;
    private String workerId;
    private String label;
    private List<String> tags;
    private List<WorkerSkillAssignment> skills;
    private List<String> teamIds;

    public InviteWorkerIdentity(String email) {
        if (email == null) {
            throw new IllegalArgumentException("email is required");
        }
        this.email = email;
    }

    public String getEmail() { return email; }
    public String getWorkerId() { return workerId; }
    public String getLabel() { return label; }
    public List<String> getTags() { return tags; }
    public List<WorkerSkillAssignment> getSkills() { return skills; }
    public List<String> getTeamIds() { return teamIds; }

    public InviteWorkerIdentity workerId(String workerId) { this.workerId = workerId; return this; }
    public InviteWorkerIdentity label(String label) { this.label = label; return this; }
    public InviteWorkerIdentity tags(List<String> tags) { this.tags = tags; return this; }
    public InviteWorkerIdentity skills(List<WorkerSkillAssignment> skills) { this.skills = skills; return this; }
    public InviteWorkerIdentity teamIds(List<String> teamIds) { this.teamIds = teamIds; return this; }

    /** Serialise for the wire, omitting unset fields. */
    public String toJson() {
        return Json.write(this);
    }
}
